// @code-inspector/turbopack/dist/index.mjs
var p = Object.defineProperty, f = Object.defineProperties;
var u = Object.getOwnPropertyDescriptors;
var n = Object.getOwnPropertySymbols;
var d = Object.prototype.hasOwnProperty, m = Object.prototype.propertyIsEnumerable;
var a = (e, r, o) => r in e ? p(e, r, { enumerable: !0, configurable: !0, writable: !0, value: o }) : e[r] = o, t = (e, r) => {
  for (var o in r || (r = {}))
    d.call(r, o) && a(e, o, r[o]);
  if (n)
    for (var o of n(r))
      m.call(r, o) && a(e, o, r[o]);
  return e;
}, s = (e, r) => f(e, u(r));
import { isDev as j } from "@code-inspector/core";
import v from "path";
import { fileURLToPath as g } from "url";
function q(e) {
  if (e.close || !j(e.dev, process.env.NODE_ENV === "development"))
    return {};
  const r = {
    port: 0,
    entry: "",
    output: e.output
  };
  let o = null;
  if (typeof require != "undefined" && typeof require.resolve == "function" && (o = require.resolve("@code-inspector/webpack")), typeof import.meta.resolve == "function") {
    const c = import.meta.resolve(
      "@code-inspector/webpack"
    );
    o = g(c);
  }
  const l = v.resolve(o, ".."), i = [
    "*.jsx",
    "*.tsx",
    "layout.js",
    "layout.ts",
    "page.js",
    "page.ts",
    "loading.js",
    "loading.ts",
    "not-found.js",
    "not-found.ts",
    "error.js",
    "error.ts",
    "global-error.js",
    "global-error.ts",
    "template.js",
    "template.ts",
    "default.js",
    "default.ts"
  ];
  return {
    ["**/{".concat(i.join(","), "}")]: {
      loaders: [
        t({
          loader: "".concat(l, "/loader.js"),
          options: s(t({}, e), {
            record: r
          })
        }, e.enforcePre === !1 ? {} : { enforce: "pre" }),
        {
          loader: "".concat(l, "/inject-loader.js"),
          options: s(t({}, e), {
            record: r
          }),
          enforce: "pre"
        }
      ]
    }
  };
}
export {
  q as TurbopackCodeInspectorPlugin
};
